File: Framework/risersoft.app.mxform.bot/Alexa/Menu/Menu.cs
using CognitiveServiceRsMx.Helpers;
using CognitiveServiceRsMx.Providers;
using Newtonsoft.Json;
using System;

namespace CognitiveServiceRsMx.Models
{

    public class Menu
	{
		[JsonProperty("week")]
		public int WeekNumber { get; set; }

		[JsonProperty("day")]
		public string Day { get; set; }
		[JsonProperty("primo")]
		public string Primo { get; set; }

		[JsonProperty("secondo")]
		public string Secondo { get; set; }
		[JsonProperty("contorno")]
		public string Contorno { get; set; }

		[JsonProperty("dolce")]
		public string Dolce { get; set; }

		public string ToString(DateTime dated, IDateTimeProvider dateTimeProvider = null)
		{
			dateTimeProvider = dateTimeProvider ?? new DateTimeProvider();

			string verbTense = null;
			if (dated == dateTimeProvider.Now().Date) {
				verbTense = "is";
			} else if (dated < dateTimeProvider.Now().Date) {
				verbTense = "was";
			} else {
				verbTense = "will be";
			}

			return string.Format("On {0} {1}, the menu {2} {3}, then {4} with {5}, finishing with {6}", Day, dated.ToStringWithSuffix("d MMMM"), verbTense, Primo, Secondo, Contorno, Dolce);
		}
	}
}

File: Framework/risersoft.app.mxform.bot/BotFx/Models/Menu.vb
Imports System
Imports CognitiveServiceRsMx.Helpers
Imports CognitiveServiceRsMx.Providers
Imports Newtonsoft.Json
Imports risersoft.app.mxform.bot.Helpers
Imports risersoft.app.mxform.bot.Providers

Namespace Models

    Public Class Menu
        <JsonProperty("week")>
        Public Property WeekNumber() As Integer

        <JsonProperty("day")>
        Public Property Day() As String
        <JsonProperty("primo")>
        Public Property Primo() As String

        <JsonProperty("secondo")>
        Public Property Secondo() As String
        <JsonProperty("contorno")>
        Public Property Contorno() As String

        <JsonProperty("dolce")>
        Public Property Dolce() As String

        Public Overloads Function ToString(dated As DateTime, Optional dateTimeProvider As IDateTimeProvider = Nothing) As String
            dateTimeProvider = If(dateTimeProvider, New DateTimeProvider())

            Dim verbTense As String
            If dated = dateTimeProvider.Now().[Date] Then
                verbTense = "is"
            ElseIf dated < dateTimeProvider.Now().[Date] Then
                verbTense = "was"
            Else
                verbTense = "will be"
            End If

            Return String.Format("On {0} {1}, the menu {2} {3}, then {4} with {5}, finishing with {6}", Day, dated.ToStringWithSuffix("d MMMM"), verbTense, Primo, Secondo, Contorno, Dolce)
        End Function
    End Class
End Namespace
